enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,413 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Documentation is updated in the upstream
+- [ ] Documentation is updated in the downstream
+
+
+## Summary
+
+Siteconfig allows customer to include custom extra manifests and list them in the siteconfig CR to be added during CR generation. It provides the flexibility to add additional CRs from the customer point of view during install time along with the CRs the RAN team provides. 
+
+The enhancement seeks to address the version independence of extra manifests, by putting version dependent manifests in version specific folder in git and pointing the directories via `extraManifestSearchPaths`.
+
+
+## Motivation
+
+* Currently customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs from git directory to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs. But it limits:
+
+    * does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+    * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+    * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+
+### Goals
+
+* Allow customer to include any custom CRs having same/different name than builtIn CRs residing in the ztp-container
+
+* Allow customer to include multiple directory path to look for extra-manifests in siteconfig, this way we are better prepared for 4.15+ features
+
+* This will enable the customer to control what CRs get added in site generation
+
+* Keep backward compatibility of the behavior corresponds to 4.13 version, meaning the new parameter is not mandatory to generate CRs through siteconfig.
+
+### Non-Goals
+
+- TBD
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-7365) --> [User Story](https://issues.redhat.com/browse/CNF-8672)
+
+## Proposal
+
+* The proposal is to use a new field as the opt-in, eg. `extraManifestSearchPaths` which can include multiple directories
+
+  ```bash
+  extraManifestSearchPaths:
+    - myPath1/
+    - myPath2/
+  ```
+
+
+  * when `extraManifestSearchPaths` is defined in Siteconfig, it will exclude any builtin CRs from ztp-container
+
+  * Going forward, we will deprecate `extraManifestPath`.
```

Comment:
```suggestion
  * We will deprecate `extraManifestPath` starting in this release. If defined it will continue to function exactly as it does today, but customers will be recommended to use the new extraManifestSearchPaths field.
```

Reply:
done